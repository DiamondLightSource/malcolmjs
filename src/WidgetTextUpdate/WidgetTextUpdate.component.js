import * as React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from 'material-ui/styles';
import Paper from 'material-ui/Paper';
import Typography from 'material-ui/Typography';
import { emphasize } from 'material-ui/styles/colorManipulator';

const styles = theme => ({
  div: {
    backgroundColor: theme.palette.background.paper,
    padding: 15,
    display: 'flex',
  },
  card: {
    minWidth: 50,
    backgroundColor: emphasize(theme.palette.background.paper, 0.2),
    padding: 2,
  },
  textUpdate: {
    textAlign: 'Center',
    marginLeft: 4,
    marginRight: 4,
  },
  textUnits: {
    alignSelf: 'Center',
    marginLeft: 4,
    marginRight: 4,
  },
});

const WidgetTextUpdate = props => (
  <div className={props.classes.div}>
    <Paper className={props.classes.card}>
      <Typography className={props.classes.textUpdate}>{props.Text}</Typography>
    </Paper>
    <Typography className={props.classes.textUnits}>
      {props.Units ? props.Units : null}
    </Typography>
  </div>
);

WidgetTextUpdate.propTypes = {
  Text: PropTypes.string.isRequired,
  Units: PropTypes.string,
  classes: PropTypes.shape({
    div: PropTypes.string,
    card: PropTypes.string,
    textUpdate: PropTypes.string,
    textUnits: PropTypes.string,
  }).isRequired,
};

WidgetTextUpdate.defaultProps = {
  Units: null,
};

export default withStyles(styles, { withTheme: true })(WidgetTextUpdate);
