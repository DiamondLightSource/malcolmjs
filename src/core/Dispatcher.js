/*
 * React.js Starter Kit
 * Copyright (c) 2014 Konstantin Tarkus (@koistya), KriaSoft LLC.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE.txt file in the root directory of this source tree.
 */

import Flux from 'flux';
import PayloadSources from '../constants/PayloadSources';
import assign from 'react/lib/Object.assign';

/**
 * A singleton that operates as the central hub for application updates.
 * For more information visit https://facebook.github.io/flux/
 */
let Dispatcher = assign(new Flux.Dispatcher(), {

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the server.
   */
  handleServerAction(action) {
    var payload = {
      source: PayloadSources.SERVER_ACTION,
      action: action
    };
    this.dispatch(payload);
  },

  /**
   * @param {object} action The details of the action, including the action's
   * type and additional data coming from the view.
   */
  handleViewAction(action) {
    var payload = {
      source: PayloadSources.VIEW_ACTION,
      action: action
    };
    this.dispatch(payload);
  }

});

export default Dispatcher;
