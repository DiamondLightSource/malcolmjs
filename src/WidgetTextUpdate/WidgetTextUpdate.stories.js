import React from 'react';

import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import WidgetTextUpdate from './WidgetTextUpdate.component';

const TextUpdate = (text, units) => (
  <WidgetTextUpdate Text={text} Units={units} />
);

storiesOf('WidgetTextUpdate', module)
  .add(
    'Hello world',
    withInfo(`
  A simple field for displaying some text.
  `)(() => TextUpdate('Hello World!', null))
  )
  .add(
    'Power level',
    withInfo(`
  A simple field for displaying some text (with units).
  `)(() => TextUpdate('1.21', 'GW'))
  );
