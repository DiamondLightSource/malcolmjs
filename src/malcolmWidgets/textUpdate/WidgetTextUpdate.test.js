import React from 'react';
import { createShallow } from '@material-ui/core/test-utils';
import WidgetTextUpdate from './WidgetTextUpdate.component';

describe('WidgetTextUpdate', () => {
  let shallow;

  beforeEach(() => {
    shallow = createShallow({ dive: true });
  });

  it('displays text', () => {
    const wrapper = shallow(<WidgetTextUpdate Text="Hello World" />);
    expect(wrapper).toMatchSnapshot();
  });

  it('displays with units', () => {
    const wrapper = shallow(<WidgetTextUpdate Text="1.21" Units="GW" />);
    expect(wrapper).toMatchSnapshot();
  });
});
