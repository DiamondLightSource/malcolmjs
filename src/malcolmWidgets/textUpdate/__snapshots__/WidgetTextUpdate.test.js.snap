// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetTextUpdate displays multiline text 1`] = `
<div
  className="WidgetTextUpdate-divNoWrap-2"
>
  <WithStyles(Typography)
    className="WidgetTextUpdate-textUpdateNoWrap-3"
    style={Object {}}
  >
    This is a slightly longer example of some text which should be split over multiple lines when the widget is in a normal sized div
  </WithStyles(Typography)>
</div>
`;

exports[`WidgetTextUpdate displays text 1`] = `
<div
  className="WidgetTextUpdate-div-1"
>
  <WithStyles(Typography)
    className="WidgetTextUpdate-textUpdate100-5"
    noWrap={true}
    style={Object {}}
  >
    Hello World
  </WithStyles(Typography)>
</div>
`;

exports[`WidgetTextUpdate displays with units 1`] = `
<div
  className="WidgetTextUpdate-div-1"
>
  <WithStyles(Typography)
    className="WidgetTextUpdate-textUpdate80-4"
    noWrap={true}
    style={Object {}}
  >
    1.21
  </WithStyles(Typography)>
  <WithStyles(Typography)
    className="WidgetTextUpdate-unitBox-6"
    noWrap={true}
  >
    GW
  </WithStyles(Typography)>
</div>
`;
